docs/checklist-command.md

```diff
@@ -0,0 +1,32 @@
+# New CLI command
+
+Subcommands should always be preferred over adding more top-level commands.
+
+Code flow for commands is generally:
+
+```
+CLI (command/) -> API Client (api/) -> HTTP API (command/agent) -> RPC (nomad/)
+```
+
+## Code
+
+1.  [ ] New file in `command/` or in an existing file if a subcommand
+2.  [ ] Test new command in `command/` package
+3.  [ ] Implement autocomplete
+4.  [ ] Implement `-json` (returns raw API response)
+5.  [ ] Implement `-verbose` (expands truncated UUIDs, adds other detail)
+7.  [ ] Update help text
+7.  [ ] Implement and test new HTTP endpoint in `command/agent/<command>_endpoint.go`
+8.  [ ] Implement and test new RPC endpoint in `nomad/<command>_endpoint.go`
+9.  [ ] Implement and test new Client RPC endpoint in
+    `client/<command>_endpoint.go` (For client endpoints like Filesystem only)
+10. [ ] Implement and test new `api/` package Request and Response structs
+11. [ ] Implement and test new `api/` package helper methods
+12. [ ] Implement and test new `nomad/structs/` package Request and Response structs
+
+## Docs
+
+1. [ ] Changelog
+2. [ ] API docs https://www.nomadproject.io/api/index.html
+3. [ ] CLI docs https://www.nomadproject.io/docs/commands/index.html
+4. [ ] Consider if it needs a guide https://www.nomadproject.io/guides/index.html
```

Comment:
Would be nice to link to `website/` directories here as well.

Reply:
I tried to avoid links to ease docs maintenance burden. Website links we add redirects for when we change, so at least they should stay fresh forever.

I want the MVP of these docs to allow devs and reviewers to track their progress on a task more than point them to exactly where each change needs to occur.

Reply:
Oh but I do mention package names everywhere, so I'll add `website/` and just not link it. That's probably what you mean all along sorry!